src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -893,7 +911,35 @@ private ZipArchiveEntry CreateEntry(ZipArchive package, string entryName, Compre
         private ZipArchiveEntry CreatePackageFileEntry(ZipArchive package, string entryName, DateTimeOffset timeOffset, CompressionLevel compressionLevel)
         {
             var entry = package.CreateEntry(entryName, compressionLevel);
-            entry.LastWriteTime = timeOffset;
+
+            // Please note: ZipArchive stream reader randomly changes LastWriteTime by another 1 second off than what "entry.LastWriteTime" has, most likely bug on their side.
+            if (timeOffset.UtcDateTime < ZipFormatMinDate)
+            {
+                _logger.Log(
+                    PackagingLogMessage.CreateMessage(
+                        string.Format(
+                            CultureInfo.CurrentCulture,
+                            Strings.ZipFileTimeStampModified, entryName, timeOffset.DateTime.ToShortDateString(), ZipFormatMinDate.ToShortDateString()),
+                        LogLevel.Information));
+
+                entry.LastWriteTime = ZipFormatMinDate;
+            }
+            else if (timeOffset.UtcDateTime > ZipFormatMaxDate)
+            {
+                _logger.Log(
+                    PackagingLogMessage.CreateMessage(
```

Comment:
this creates n messages, you can consider printing only/exactly one message instead.

Reply:
Ok. Addressed.